src/solc_0.8/StarterPack/StarterPackV2.sol

```diff
@@ -0,0 +1,382 @@
+    }
+
+    function _isValidCatalyst(uint16 catalystId) internal view returns (bool) {
+        return GemsCatalystsRegistry(_registry).doesCatalystExist(catalystId) && catalystId > 0;
+    }
+
+    function _getGem(uint16 gemId) internal view returns (IGem) {
+        return GemsCatalystsRegistry(_registry).getGem(gemId);
+    }
+
+    function _isValidGem(uint16 gemId) internal view returns (bool) {
+        return GemsCatalystsRegistry(_registry).doesGemExist(gemId) && gemId > 0;
+    }
+
+    /// @dev Function to calculate the total price in SAND of the StarterPacks to be purchased
+    function _calculateTotalPriceInSAND(
+        uint256[] memory catalystIds,
+        uint256[] memory catalystQuantities,
+        uint256[] memory gemIds,
+        uint256[] memory gemQuantities
+    ) internal returns (uint256) {
+        require(catalystIds.length == catalystQuantities.length, "INVALID_CAT_INPUT");
+        require(gemIds.length == gemQuantities.length, "INVALID_GEM_INPUT");
+        bool useCurrentPrices = _priceSelector();
+        uint256 totalPrice;
+        for (uint256 i = 0; i < catalystIds.length; i++) {
+            uint16 id = uint16(catalystIds[i]);
+            uint256 quantity = catalystQuantities[i];
+            totalPrice =
+                totalPrice +
+                (useCurrentPrices ? _catalystPrices[id] * (quantity) : _catalystPreviousPrices[id] * (quantity));
+        }
+        for (uint256 i = 0; i < gemIds.length; i++) {
+            uint16 id = uint16(gemIds[i]);
+            uint256 quantity = gemQuantities[i];
+            totalPrice =
+                totalPrice +
+                (useCurrentPrices ? _gemPrices[id] * (quantity) : _gemPreviousPrices[id] * (quantity));
+        }
+        return totalPrice;
+    }
+
+    /// @dev Function to determine whether to use previous or current prices
+    function _priceSelector() internal returns (bool) {
+        bool useCurrentPrices;
+        // No price change
+        if (_priceChangeTimestamp == 0) {
+            useCurrentPrices = true;
+        } else {
+            // Price change delay has expired: use current prices
+            if (block.timestamp > _priceChangeTimestamp + PRICE_CHANGE_DELAY) {
+                _priceChangeTimestamp = 0;
+                useCurrentPrices = true;
+            } else {
+                // Price change has recently occured: use previous prices until price change takes effect
+                useCurrentPrices = false;
+            }
+        }
+        return (useCurrentPrices);
+    }
+
+    /// @dev Function to handle purchase with SAND
+    function _transferSANDPayment(
+        address buyer,
+        address payable paymentRecipient,
+        uint256 amount
+    ) internal {
+        uint256 amountForDestination = amount;
+        require(IERC20(_sand).transferFrom(buyer, paymentRecipient, amountForDestination), "PAYMENT_TRANSFER_FAILED");
+    }
+
+    /// @dev this override is required
```

Comment:
I the overrides are excluded I get the following errors (because I'm inheriting ../common/BaseWithStorage/ERC2771Handler.sol):
`TypeError: Derived contract must override function "_msgSender". Two or more base classes define function with same name and parameter types.`
`TypeError: Derived contract must override function "_msgData". Two or more base classes define function with same name and parameter types`